### ISSUE TITLE
Investigate parentage of StepChain saving multi-steps output

### ISSUE THREAD

**Issue by amaltaro** [dmwm/WMCore#8269]
Since the merge jobs happen in parallel, I'm not sure their parentage is set properly. E.g.:
Step1: with input data, drop output
Step2: save output (parent is the input dataset)
Step3: save output (parent has to be output of Step2) --> though I think it uses the input dataset instead.

**Comment by ticoann** [comment]
I just talked Yuyi, it seems DBS will catch this situation and throw and error.
If the parent file is not in DBS and child file was getting inserted with that parent file DBS insert fails for the whole block. @yuyiguo, Yuyi, that is correct, right?

**Comment by amaltaro** [comment]
I've just had a look at one case and it's indeed broken when we stage out output from multiple steps. It simply uses the input dataset as parent of all the output datasets. I'll have to think hard on how to fix it.

**Comment by vlimant** [comment]
NB. this is the only road block between stepchain and production.
We have ~nothing in production that produces only output from the one step.

**Comment by ticoann** [comment]
Closing this and create a new issue as we discussed. #8556
